### PR TITLE
More resilient ILM history rollover test

### PR DESCRIPTION
**Pull request by probakowski**
This adds some protection in ILM history rollover test to remove its flakiness

### CODE REVIEW

File: x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/ILMHistoryTests.java [hunk 1]
```diff
@@ -91,8 +91,12 @@ public void testIlmHistoryIndexCanRollover() throws Exception {
 
         //wait for all history items to index to avoid waiting for timeout in ILMHistoryStore beforeBulk
         assertBusy(() -> {
-            SearchResponse search = client().prepareSearch(firstIndex).setQuery(matchQuery("index", firstIndex)).setSize(0).get();
-            assertThat(search.getHits().getTotalHits().value, is(9L));
+            try {
+                SearchResponse search = client().prepareSearch(firstIndex).setQuery(matchQuery("index", firstIndex)).setSize(0).get();
+                assertThat(search.getHits().getTotalHits().value, is(9L));
+            } catch (Exception e) {
+                fail(e.getMessage());
```
Comment (andreidan) [inline x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/ILMHistoryTests.java:98]: Thanks for fixing this Przemko.

Can you add a comment on why this is needed please? it's not obvious `assertBusy` fails on first non-AssertionError exception. We could also use `assertHitCount` maybe? 

Is it worth also increasing the timeout of `assertBusy` to 30 seconds?

Comment (probakowski) [reply x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/ILMHistoryTests.java:101]: I've added a comment and switched to `assertHitCount`. I don't think we need to increase timeout - we should have to wait for at most 5s for all ILM history items to be index

**Comment by probakowski** [comment]
@elasticmachine update branch

**Review by andreidan — APPROVED** [review]
LGTM

**Comment by probakowski** [comment]
This was backported to `7.x` as part of #61661

**Pull request merged**